sample: add redirection README for gkeonprem samples

Shabirmean opened the pull request:
### Description
- This PR adds a new folder for the **gkeonprem_** resource of the Google Terraform provider
- The folder is intended to host the samples for the Anthos onPrem clusters using the **gkeonprem_** resource variant.
- However, these samples are hosted in [go/anthos-samples](http://go/anthos-samples) where all the other Anthos specific samples are hosted. 
- Thus, we place a placeholder folder in this repository and refer to the location in the [go/anthos-samples](http://go/anthos-samples)  repository where the samples can be found using a README file.

### Note:
- The PR must be merged only after [PR#593](https://github.com/GoogleCloudPlatform/anthos-samples/pull/593) of the anthos-samples repo has been merged.

### Related PR
- #325
- https://github.com/terraform-google-modules/terraform-docs-samples/pull/397

Shabirmean commented:
I get the following error when I run this sample:
- **(Testing on ABM 1.14.2)**:

```
│ Error: Error creating BareMetalCluster: googleapi: Error 400: Invalid JSON payload received. Unknown name "nodePoolConfig" at 'bare_metal_cluster.control_plane': Cannot find field.
│ Details:
│ [
│   {
│     "@type": "type.googleapis.com/google.rpc.BadRequest",
│     "fieldViolations": [
│       {
│         "description": "Invalid JSON payload received. Unknown name \"nodePoolConfig\" at 'bare_metal_cluster.control_plane': Cannot find field.",
│         "field": "bare_metal_cluster.control_plane"
│       }
│     ]
│   }
│ ]
│ 
│   with google_gkeonprem_bare_metal_cluster.gkeonprem-bm-cluster-1,
│   on main.tf line 42, in resource "google_gkeonprem_bare_metal_cluster" "gkeonprem-bm-cluster-1":
│   42: resource "google_gkeonprem_bare_metal_cluster" "gkeonprem-bm-cluster-1" {
```

Need to check with Terraform Eng team to see what's the root of the issue!

msampathkumar commented:
This PR looks like a duplicate PR of #325?

Kindly update PR title or close duplicates.

betsy-lichtenberg commented:
/gcbrun

Shabirmean commented:
@msampathkumar @betsy-lichtenberg - please TAL when you have some time!

betsy-lichtenberg commented:
/gcbrun

betsy-lichtenberg commented on gkeonprem/gkeonprem_baremetal_cluster_metallb/main.tf:
```diff
@@ -0,0 +1,155 @@
+# already have an admin cluster created according to the prerequisites outlined
+# in https://cloud.google.com/anthos/clusters/docs/bare-metal/latest/installing/creating-clusters/create-user-cluster-api#before_you_begin
+locals {
+  project           = "YOUR_PROJECT_ID"
```
Please delete the `project` argument or comment it out.

We direct users to use `export GOOGLE_CLOUD_PROJECT` instead.

see https://cloud.google.com/docs/terraform/basic-commands#google_project

betsy-lichtenberg replied:
Delete all occurrences:

`project                   = local.project`

betsy-lichtenberg replied:
Where necessary, you can use `data "google_project"` instead

betsy-lichtenberg replied:
Yes, the writer [guidelines](https://cloud.google.com/guides/authoring/terraform/include-on-cgc#include-instructions) say to include or link to the document from the docs. 

The way the CI test framework works for this repo, they spin up a new project every time, so specifying a project doesn't work. 

Can you comment out the project ID? If not, you can try adding a test.yaml file to skip testing.

betsy-lichtenberg replied:
Please see the contributing guide for more information about testing.

betsy-lichtenberg replied:
See also: https://cloud.google.com/guides/authoring/terraform/include-on-cgc#include-instructions

msampathkumar replied:
@Shabirmean - please also update `local.project` suggestion, for PR https://github.com/terraform-google-modules/terraform-docs-samples/pull/397/files

Shabirmean replied:
Cool. Thank you. I don't think we can run integration tests on these samples. They are targeted towards a specific baremetal infrastructure. So we either need to have one setup or emulate one using GCE VMs to be able to test them. Thus, I have added a skip yaml file (test.yaml).

Also regarding the project id, I have added the `data resource` with an attribute and have added comments about it. Let me know wdyt!

betsy-lichtenberg commented:
https://pantheon.corp.google.com/cloud-build/builds/aeba06ae-04d5-466f-9e17-f808b9b84998?project=cloud-foundation-cicd

Error: Trailing whitespace found in the lines above.

Checking for missing newline at end of file
Error: No newline at end of file ./compute/os_config_policy/main.tf
Error: No newline at end of file ./.github/CODEOWNERS
Running shellcheck

Shabirmean commented:
> https://pantheon.corp.google.com/cloud-build/builds/aeba06ae-04d5-466f-9e17-f808b9b84998?project=cloud-foundation-cicd
> 
> Error: Trailing whitespace found in the lines above.
> 
> Checking for missing newline at end of file Error: No newline at end of file ./compute/os_config_policy/main.tf Error: No newline at end of file ./.github/CODEOWNERS Running shellcheck

I have fixed them, but these are not files I touched. These seems to be errors from before!

betsy-lichtenberg commented on gkeonprem/gkeonprem_baremetal_cluster_metallb/main.tf:
```diff
@@ -0,0 +1,160 @@
+# already have an admin cluster created according to the prerequisites outlined
+# in https://cloud.google.com/anthos/clusters/docs/bare-metal/latest/installing/creating-clusters/create-user-cluster-api#before_you_begin
+locals {
+  admin_cluster     = "YOUR_ADMIN_CLUSTER_NAME"
```
Can you reverse this where you put the values in the quotes and the meaning of the value in the comments?

My hope is that by doing this the sample will run as-is and customers can replace the hard-coded values with their own values.

Shabirmean commented:
@betsy-lichtenberg Please TAL. I have added just the README and moved the sample to [go/anthos-samples](http://go/anthos-samples)  as discussed offline.

betsy-lichtenberg commented on gkeonprem/test.yaml:
```diff
@@ -0,0 +1,20 @@
+# Copyright 2023 Google LLC
```
Pretty sure you can delete this file.
